airflow/providers/google/cloud/operators/cloud_storage_transfer_service.py

```diff
@@ -861,6 +867,10 @@ def __init__(
     def _validate_inputs(self) -> None:
         if self.delete_job_after_completion and not self.wait:
             raise AirflowException("If 'delete_job_after_completion' is True, then 'wait' must also be True.")
+        if self.gcs_path and not self.gcs_path.endswith("/"):
+            raise AirflowException(
+                "The destination Google Cloud Storage path must end with a slash '/' or be empty."
+            )
```

Comment:
I know nothing about GCS, but if the path must ends with `/`, can we automatically append it instead of failing here?

Reply:
Thank you, makes sense!
I've checked if any similar function already exists and found the one in the `GCSHook` class [here](https://github.com/apache/airflow/blob/0f97b92c1ad15bd6d0a90c8dee8287886641d7d9/airflow/providers/google/cloud/hooks/gcs.py#L1140)
To avoid repetitive code I've extracted class function into a standalone one in order to import it here.
Please let me know your thoughts.